### PR TITLE
Add support for ssh:// and unix:// podman clients

### PR DESCRIPTION
**Pull request by jwhonce**
* Make context keys package safe
 * Add support for PODMAN_HOST and PODMAN_SSHKEY
 * Add slight increasing delay when client connections fail
 * Remove usages of path.Join(), added JoinURL(). '/' is not OS
   dependent.
 * Add support for secure connections to server

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: pkg/bindings/connection.go [hunk 1]
```diff
@@ -85,26 +135,40 @@ func pingNewConnection(ctx context.Context) error {
...
 
-// newConnection takes a scheme and address and creates a connection from it
-func newConnection(scheme, address string) Connection {
-	client := http.Client{
+func sshClient(_url *url.URL, identity string) (*http.Client, error) {
+	auth, err := publicKey(identity)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to parse identity %s: %v\n", identity)
+	}
+	bastion, err := ssh.Dial("tcp",
```
Comment (AkihiroSuda) [inline pkg/bindings/connection.go]: This doesn't support `~/.ssh/config` and yet `InsecureIgnoreHostKey()` is insecure.
I suggest porting over `docker system dial-stdio`.

**Comment by rhatdan** [comment]
@afbjorklund PTAL

File: pkg/bindings/connection.go (hunk 1 above)
Comment (jwhonce) [reply pkg/bindings/connection.go]: @AkihiroSuda We trying to actively avoid running ssh on the client.  We did that in the previous version of the API and it was very painful for users. I can work on not using `InsecureIgnoreHostKey()`, or at least make it optional.

Comment (AkihiroSuda) [reply pkg/bindings/connection.go]: Without using ssh binary, isn't it hard to support bastions?

Comment (jwhonce) [reply pkg/bindings/connection.go]: sshd is only required on the bastion, no ssh binary on the client.  I updated the code to allow for known_hosts and secure connections.

Comment (AkihiroSuda) [reply pkg/bindings/connection.go]: Sorry, I meant `ProxyCommand` stuff

**Comment by baude** [comment]
lgtm

**Pull request merged**